core/capabilities/ccip/ocrimpls/evm_contract_transmitter_factory.go

```diff
@@ -0,0 +1,98 @@
+package ocrimpls
+
+import (
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3types"
+	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+
+	"github.com/smartcontractkit/chainlink-ccip/pkg/consts"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+
+	ccipcommon "github.com/smartcontractkit/chainlink/v2/core/capabilities/ccip/common"
+)
+
+// EVMContractTransmitterFactory implements ContractTransmitterFactory for EVM-based chains.
+type EVMContractTransmitterFactory struct{}
+
+var EVMExecCallDataFunc = func(
+	rawReportCtx [2][32]byte,
+	report ocr3types.ReportWithInfo[[]byte],
+	_, _ [][32]byte,
+	_ [32]byte,
+	_ ccipcommon.ExtraDataCodec,
+) (contract string, method string, args any, err error) {
+	// Note that the name of the struct field is very important, since the encoder used
+	// by the chainwriter uses mapstructure, which will use the struct field name to map
+	// to the argument name in the function call.
+	// If, for whatever reason, we want to change the field name, make sure to add a `mapstructure:"<arg_name>"` tag
+	// for that field.
+
+	return consts.ContractNameOffRamp,
+		consts.MethodExecute,
+		struct {
+			ReportContext [2][32]byte
+			Report        []byte
+		}{
+			ReportContext: rawReportCtx,
+			Report:        report.Report,
+		}, nil
+}
+
+// EVMCommitCalldataFunc returns a ToCalldataFunc that omits the Info object for EVM.
+func EVMCommitCalldataFunc(defaultMethod string) ToCalldataFunc {
```

Comment:
```suggestion
func EVMCommitCalldataFunc(commitMethod string) ToCalldataFunc {
```

`default` makes no sense in this context